src/components/AddRecipe.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import RecipeService from '../services/RecipeService';
+
+const AddRecipe = () => {
+  const initialRecipeState = {
+    id: null,
+    name: '',
+    description: '',
+    ingredients: '',
+  };
+
+  const [recipe, setRecipe] = useState(initialRecipeState);
+  const [submitted, setSubmitted] = useState(false);
+
+  const handleInputChange = event => {
+    const { name, value } = event.target;
+    setRecipe({ ...recipe, [name]: value });
+  };
+
+  const saveRecipe = () => {
+    const ingredients = recipe.ingredients
+      .split(',')
+      .map(ingredient => ({ name: ingredient.trim() }));
+
+    const data = {
+      name: recipe.name,
+      description: recipe.description,
+      ingredients: ingredients,
+    };
+
+    RecipeService.create(data)
+      .then(response => {
+        setRecipe({
+          id: response.data.id,
+          name: response.data.name,
+          description: response.data.description,
+          ingredients: response.data.ingredients,
+        });
+
+        setSubmitted(true);
+        console.log(response.data);
```

Comment:
Please let's avoid using console.log for code that would be merged to the master branch. This should only be used for local testing. In this exercise, if we want to provide feedback to the user, `alert` would be a better option since that would be visible in the screen. Also, this is polluting the output of the test.